## Extracción de la información

In [40]:
import yfinance as yf
import pandas as pd
import matplotlib.pyplot as plt
from datetime import datetime, timedelta
import numpy as np
import os
import warnings

# Suprimir todos los FutureWarnings
warnings.filterwarnings('ignore', category=FutureWarning)

# Suprimir warnings específicos de yfinance
warnings.filterwarnings('ignore', message='.*auto_adjust.*')

pd.options.mode.chained_assignment = None

# Descargar datos de BBVA

bbva = yf.download('BBVA.MC', start='2010-01-01', end='2025-08-31')
santander = yf.download('SAN.MC', start='2010-01-01', end='2025-08-31')
sabadell = yf.download('SAB.MC', start='2010-01-01', end='2025-08-31')
bankinter = yf.download('BKT.MC', start='2010-01-01', end='2025-08-31')
caixabank = yf.download('CABK.MC', start='2010-01-01', end='2025-08-31')
unicaja = yf.download('UNI.MC', start='2010-01-01', end='2025-08-31')


print(bbva.describe(),"\n") # Aplicar lo aprendido en Summary Function de curso Kaggle
print(santander.describe(),"\n")
print(bankinter.describe(), "\n")
print(sabadell.describe(),"\n")
print(caixabank.describe(),"\n")
print(unicaja.describe(),"\n")



[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed
[*********************100%***********************]  1 of 1 completed


Price         Close         High          Low         Open        Volume
Ticker      BBVA.MC      BBVA.MC      BBVA.MC      BBVA.MC       BBVA.MC
count   4007.000000  4007.000000  4007.000000  4007.000000  4.007000e+03
mean       4.721315     4.778106     4.657952     4.719907  3.935376e+07
std        2.266704     2.282805     2.242114     2.260112  4.295000e+07
min        1.646662     1.696977     1.622267     1.639039  0.000000e+00
25%        3.470244     3.518139     3.423629     3.475497  1.627335e+07
50%        4.076172     4.131225     4.031005     4.080708  2.746292e+07
75%        4.932289     4.982468     4.874550     4.929974  4.618471e+07
max       16.480000    16.700001    16.445000    16.629999  6.009019e+08 

Price         Close         High          Low         Open        Volume
Ticker       SAN.MC       SAN.MC       SAN.MC       SAN.MC        SAN.MC
count   4007.000000  4007.000000  4007.000000  4007.000000  4.007000e+03
mean       3.405571     3.448572     3.360051    

## Analisis previo

In [41]:
# Mostramos las primeras 10 líneas

print(bbva.head(10), "\n")
print(sabadell.head(10), "\n")
print(santander.head(10), "\n")
print(unicaja.head(10), "\n")
print(bankinter.head(10), "\n")
print(caixabank.head(10), "\n")

Price          Close      High       Low      Open    Volume
Ticker       BBVA.MC   BBVA.MC   BBVA.MC   BBVA.MC   BBVA.MC
Date                                                        
2010-01-04  5.410954  5.410954  5.306656  5.310828  91911560
2010-01-05  5.448501  5.471447  5.406783  5.410954  48280066
2010-01-06  5.469361  5.492307  5.425557  5.448502  12069848
2010-01-07  5.444328  5.473532  5.385922  5.469359  21497189
2010-01-08  5.486049  5.513166  5.435986  5.444330  44302415
2010-01-11  5.448501  5.536111  5.435986  5.486049  42657631
2010-01-12  5.446414  5.494391  5.419297  5.448500  31361471
2010-01-13  5.402610  5.463102  5.388008  5.446415  37275170
2010-01-14  5.410954  5.450587  5.362977  5.402610  25719625
2010-01-15  5.344204  5.477705  5.315001  5.410955  37230304 

Price          Close      High       Low      Open    Volume
Ticker        SAB.MC    SAB.MC    SAB.MC    SAB.MC    SAB.MC
Date                                                        
2010-01-04  1.328521  

Vemos que la información que nos muestran son los precios y no la rentabilidad que aportan las acciones de cada entidad

# Significado de los datos de cada columna:

* Date: Fecha de negociación.
* Open: Precio al que abre la acción ese día.
* High: Precio máximo alcanzado durante el día.
* Low: Precio mínimo del día.
* Close: Precio al cierre del mercado (dato por el que nos guiaremos para el estudio)
* Volume: Número de acciones negociadas ese día.

In [42]:
# Revisamos si tienes datos nulos
print(bbva.info(), "\n")
print(sabadell.info(), "\n")
print(santander.info(), "\n")
print(unicaja.info(), "\n")
print(bankinter.info(), "\n")
print(caixabank.info(), "\n")

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 4007 entries, 2010-01-04 to 2025-08-29
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   (Close, BBVA.MC)   4007 non-null   float64
 1   (High, BBVA.MC)    4007 non-null   float64
 2   (Low, BBVA.MC)     4007 non-null   float64
 3   (Open, BBVA.MC)    4007 non-null   float64
 4   (Volume, BBVA.MC)  4007 non-null   int64  
dtypes: float64(4), int64(1)
memory usage: 187.8 KB
None 

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 4007 entries, 2010-01-04 to 2025-08-29
Data columns (total 5 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   (Close, SAB.MC)   4007 non-null   float64
 1   (High, SAB.MC)    4007 non-null   float64
 2   (Low, SAB.MC)     4007 non-null   float64
 3   (Open, SAB.MC)    4007 non-null   float64
 4   (Volume, SAB.MC)  4007 non-null   int64  
dtypes: float64(4), int64(1)

Podemos observar que no hay datos nulos, pero en el caso de Unicaja solo tenemos 2089 registros con respecto a los 4007 registros del resto de entidades. Esta diferencia se debe porque Unicaja entró en bolsa el 30 de junio del 2017.

## INFORMACION CLAVE PARA ANALISIS BANCARIO
# Datos de precios (información actual extraída de yfinance):
* Open/Close/High/Low: Actividad diaria y volatilidad intradiaria
* Volume: Liquidez y la actividad del mercado
* Fechas: Tendencias temporales y estacionalidad

# Indicadores Técnicos (calculables con los datos actuales):
* Rentabilidad: Periodicidad diaria, mensual o anual
* Volatilidad: Riesgo del activo
* Medias móviles: Tendencias de corto/largo plazo
* RSI: Sobrecompra/sobreventa
* Correlaciones: Cómo se mueven juntos los bancos

In [44]:
# Cálculo de la rentabilidad

def calcular_rentabilidades_detallado(datos, nombre_banco):
    """
    Calcula diferentes tipos de rentabilidades paso a paso - VERSIÓN CORREGIDA
    """
    print(f"\n=== RENTABILIDADES DE {nombre_banco} ===")
    
    # Manejar datos MultiIndex
    if len(datos.columns.levels) > 1:
        precios = datos['Close'].iloc[:, 0]  # Primera columna
    else:
        precios = datos['Close']
    
    # 1.1 Rentabilidad diaria
    rentabilidad_diaria = precios.pct_change() * 100
    print(f"Rentabilidad diaria promedio: {float(rentabilidad_diaria.mean()):.3f}%")
    print(f"Mejor día: {float(rentabilidad_diaria.max()):.2f}%")
    print(f"Peor día: {float(rentabilidad_diaria.min()):.2f}%")
    
    # 1.2 Rentabilidad mensual
    datos_mensuales = precios.resample('M').last()
    rentabilidad_mensual = datos_mensuales.pct_change() * 100
    print(f"Rentabilidad mensual promedio: {float(rentabilidad_mensual.mean()):.2f}%")
    
    # 1.3 Rentabilidad anual
    datos_anuales = precios.resample('Y').last()
    rentabilidad_anual = datos_anuales.pct_change() * 100
    print(f"Rentabilidad anual promedio: {float(rentabilidad_anual.mean()):.2f}%")
    
    # 1.4 Rentabilidad acumulada total
    precio_inicial = float(precios.iloc[0])
    precio_final = float(precios.iloc[-1])
    rentabilidad_total = ((precio_final / precio_inicial) - 1) * 100
    print(f"Rentabilidad total del período: {rentabilidad_total:.2f}%")
    
    # 1.5 Rentabilidad anualizada
    años = len(precios) / 252  # 252 días hábiles por año
    rentabilidad_anualizada = ((precio_final / precio_inicial) ** (1/años) - 1) * 100
    print(f"Rentabilidad anualizada: {rentabilidad_anualizada:.2f}%")
    
    return {
        'diaria': rentabilidad_diaria,
        'mensual': rentabilidad_mensual,
        'anual': rentabilidad_anual,
        'total': rentabilidad_total,
        'anualizada': rentabilidad_anualizada
    }

In [45]:
# Cálculo de la volatilidad

def calcular_volatilidad_detallado(datos, nombre_banco):
    """
    Calcula diferentes medidas de volatilidad - VERSIÓN CORREGIDA
    """
    print(f"\n=== VOLATILIDAD DE {nombre_banco} ===")
    
    # Manejar datos MultiIndex
    if len(datos.columns.levels) > 1:
        precios = datos['Close'].iloc[:, 0]
        high_prices = datos['High'].iloc[:, 0]
        low_prices = datos['Low'].iloc[:, 0]
    else:
        precios = datos['Close']
        high_prices = datos['High']
        low_prices = datos['Low']
    
    # 2.1 Volatilidad basada en rentabilidades diarias
    rentabilidades = precios.pct_change().dropna()
    volatilidad_diaria = float(rentabilidades.std()) * 100
    print(f"Volatilidad diaria: {volatilidad_diaria:.3f}%")
    
    # 2.2 Volatilidad anualizada (método estándar)
    volatilidad_anualizada = volatilidad_diaria * np.sqrt(252)
    print(f"Volatilidad anualizada: {volatilidad_anualizada:.2f}%")
    
    # 2.3 Volatilidad rolling (ventana móvil de 30 días)
    volatilidad_rolling_30 = rentabilidades.rolling(window=30).std() * np.sqrt(252) * 100
    print(f"Volatilidad rolling 30 días (última): {float(volatilidad_rolling_30.iloc[-1]):.2f}%")
    
    # 2.4 Volatilidad basada en High-Low (alternativa)
    volatilidad_high_low = float(((high_prices - low_prices) / precios).mean()) * 100
    print(f"Volatilidad High-Low promedio: {volatilidad_high_low:.2f}%")
    
    return {
        'diaria': volatilidad_diaria,
        'anualizada': volatilidad_anualizada,
        'rolling_30': volatilidad_rolling_30,
        'high_low': volatilidad_high_low
    }

In [46]:
# Cálculo de medias móviles

def calcular_medias_moviles_detallado(datos, nombre_banco):
    """
    Calcula diferentes medias móviles y señales - VERSIÓN CORREGIDA
    """
    print(f"\n=== MEDIAS MÓVILES DE {nombre_banco} ===")
    
    # Manejar datos MultiIndex
    if len(datos.columns.levels) > 1:
        precios = datos['Close'].iloc[:, 0]
    else:
        precios = datos['Close']
    
    # 3.1 Medias móviles simples (SMA)
    sma_10 = precios.rolling(window=10).mean()
    sma_20 = precios.rolling(window=20).mean()
    sma_50 = precios.rolling(window=50).mean()
    sma_200 = precios.rolling(window=200).mean()
    
    # 3.2 Medias móviles exponenciales (EMA)
    ema_12 = precios.ewm(span=12).mean()
    ema_26 = precios.ewm(span=26).mean()
    
    # 3.3 Valores actuales (convertir a float)
    precio_actual = float(precios.iloc[-1])
    sma_20_actual = float(sma_20.iloc[-1])
    sma_50_actual = float(sma_50.iloc[-1])
    sma_200_actual = float(sma_200.iloc[-1])
    
    print(f"Precio actual: {precio_actual:.2f}€")
    print(f"SMA 20: {sma_20_actual:.2f}€")
    print(f"SMA 50: {sma_50_actual:.2f}€")
    print(f"SMA 200: {sma_200_actual:.2f}€")
    
    # 3.4 Señales técnicas
    if precio_actual > sma_20_actual > sma_50_actual:
        print("Señal: TENDENCIA ALCISTA (precio > SMA20 > SMA50)")
    elif precio_actual < sma_20_actual < sma_50_actual:
        print("Señal: TENDENCIA BAJISTA (precio < SMA20 < SMA50)")
    else:
        print("Señal: LATERAL (señales mixtas)")
    
    # 3.5 Cruce de medias (Golden Cross / Death Cross)
    sma_50_anterior = float(sma_50.iloc[-5])
    sma_200_anterior = float(sma_200.iloc[-5])
    
    if sma_50_actual > sma_200_actual and sma_50_anterior <= sma_200_anterior:
        print("GOLDEN CROSS: SMA50 cruzó por encima de SMA200 (ALCISTA)")
    elif sma_50_actual < sma_200_actual and sma_50_anterior >= sma_200_anterior:
        print("DEATH CROSS: SMA50 cruzó por debajo de SMA200 (BAJISTA)")
    
    return {
        'SMA_10': sma_10,
        'SMA_20': sma_20,
        'SMA_50': sma_50,
        'SMA_200': sma_200,
        'EMA_12': ema_12,
        'EMA_26': ema_26
    }

In [47]:
# Cálculo de RSI (Relative Strength Index)

def calcular_rsi_detallado(datos, nombre_banco, periodo=14):
    """
    Calcula RSI paso a paso con interpretación - VERSIÓN CORREGIDA
    """
    print(f"\n=== RSI DE {nombre_banco} ===")
    
    # Manejar datos MultiIndex
    if len(datos.columns.levels) > 1:
        precios = datos['Close'].iloc[:, 0]
    else:
        precios = datos['Close']
    
    # 4.1 Calcular diferencias de precios
    delta = precios.diff()
    
    # 4.2 Separar ganancias y pérdidas
    ganancias = delta.where(delta > 0, 0)
    perdidas = -delta.where(delta < 0, 0)
    
    # 4.3 Medias móviles de ganancias y pérdidas
    media_ganancias = ganancias.rolling(window=periodo).mean()
    media_perdidas = perdidas.rolling(window=periodo).mean()
    
    # 4.4 Calcular RS (Relative Strength)
    rs = media_ganancias / media_perdidas
    
    # 4.5 Calcular RSI
    rsi = 100 - (100 / (1 + rs))
    
    # 4.6 Interpretación del RSI actual
    rsi_actual = float(rsi.iloc[-1])
    print(f"RSI actual: {rsi_actual:.1f}")
    
    if rsi_actual > 70:
        print("Interpretación: SOBRECOMPRADO (considerar venta)")
    elif rsi_actual < 30:
        print("Interpretación: SOBREVENDIDO (considerar compra)")
    else:
        print("Interpretación: NEUTRAL (rango normal)")
    
    # 4.7 Estadísticas del RSI
    rsi_promedio = float(rsi.mean())
    veces_sobrecomprado = (rsi > 70).sum()
    veces_sobrevendido = (rsi < 30).sum()
    
    print(f"RSI promedio: {rsi_promedio:.1f}")
    print(f"Veces sobrecomprado (>70): {veces_sobrecomprado} días")
    print(f"Veces sobrevendido (<30): {veces_sobrevendido} días")
    
    return rsi

In [48]:
# Cálculo de las correlaciones

def calcular_correlaciones_detallado(datos_bancos):
    """
    Calcula matriz de correlaciones y análisis - VERSIÓN CORREGIDA
    """
    print(f"\n=== CORRELACIONES ENTRE BANCOS ===")
    
    # 5.1 Crear DataFrame con precios de cierre
    precios = pd.DataFrame()
    for nombre, datos in datos_bancos.items():
        # Manejar datos MultiIndex
        if len(datos.columns.levels) > 1:
            precios[nombre] = datos['Close'].iloc[:, 0]
        else:
            precios[nombre] = datos['Close']
    
    # 5.2 Calcular rentabilidades
    rentabilidades = precios.pct_change().dropna()
    
    # 5.3 Matriz de correlación
    correlaciones = rentabilidades.corr()
    print("Matriz de correlaciones:")
    print(correlaciones.round(3))
    
    # 5.4 Interpretaciones
    print(f"\nInterpretaciones:")
    for i in range(len(correlaciones.columns)):
        for j in range(i+1, len(correlaciones.columns)):
            banco1 = correlaciones.columns[i]
            banco2 = correlaciones.columns[j]
            corr_value = float(correlaciones.iloc[i, j])
            
            if corr_value > 0.8:
                nivel = "MUY ALTA"
            elif corr_value > 0.6:
                nivel = "ALTA"
            elif corr_value > 0.3:
                nivel = "MODERADA"
            else:
                nivel = "BAJA"
            
            print(f"{banco1} vs {banco2}: {corr_value:.3f} ({nivel})")
    
    # 5.5 Promedio de correlaciones (riesgo sistémico)
    correlacion_promedio = float(correlaciones.values[np.triu_indices_from(correlaciones.values, k=1)].mean())
    print(f"\nCorrelación promedio del sector: {correlacion_promedio:.3f}")
    
    if correlacion_promedio > 0.7:
        print("RIESGO SISTÉMICO ALTO: Los bancos se mueven muy similares")
    elif correlacion_promedio > 0.5:
        print("RIESGO SISTÉMICO MODERADO")
    else:
        print("RIESGO SISTÉMICO BAJO: Buena diversificación")
    
    return correlaciones, rentabilidades

In [49]:
# Función para el análisis

def analisis_completo_indicadores():
    """
    Ejecuta análisis completo con tus datos existentes
    """
    # Usar tus datos ya cargados
    bancos_datos = {
        'BBVA': bbva,
        'Santander': santander,
        'CaixaBank': caixabank,
        'Sabadell': sabadell,
        'Bankinter': bankinter,
        'Unicaja': unicaja
    }
    
    # Análisis individual para cada banco
    for nombre, datos in bancos_datos.items():
        print(f"\n{'='*60}")
        print(f"ANÁLISIS TÉCNICO COMPLETO: {nombre}")
        print(f"{'='*60}")
        
        # Calcular todos los indicadores
        rentabilidades = calcular_rentabilidades_detallado(datos, nombre)
        volatilidad = calcular_volatilidad_detallado(datos, nombre)
        medias = calcular_medias_moviles_detallado(datos, nombre)
        rsi = calcular_rsi_detallado(datos, nombre)
    
    # Análisis de correlaciones (todos los bancos juntos)
    correlaciones, rentabilidades = calcular_correlaciones_detallado(bancos_datos)
    
    print(f"\n{'='*60}")
    print("ANÁLISIS COMPLETADO")
    print(f"{'='*60}")

In [50]:
# Ejemplo con BBVA

def ejemplo_rapido_bbva():
    """
    Ejemplo rápido con BBVA para entender los cálculos - SIN ERRORES DE FORMATO
    """
    print("=== EJEMPLO RÁPIDO CON BBVA ===")
    
    # Acceder correctamente a datos MultiIndex
    if len(bbva.columns.levels) > 1:
        # Datos MultiIndex
        precios = bbva['Close'].iloc[:, 0]  # Primera columna de Close
    else:
        # Datos normales
        precios = bbva['Close']
    
    # Rentabilidad diaria
    rent_diaria = precios.pct_change() * 100
    rent_promedio = float(rent_diaria.mean())
    print(f"Rentabilidad diaria promedio BBVA: {rent_promedio:.3f}%")
    
    # Volatilidad anualizada
    volatilidad = float(rent_diaria.std() * np.sqrt(252))
    print(f"Volatilidad anualizada BBVA: {volatilidad:.2f}%")
    
    # SMA 50
    sma50 = float(precios.rolling(50).mean().iloc[-1])
    print(f"SMA50 actual BBVA: {sma50:.2f}€")
    
    # Precio actual
    precio_actual = float(precios.iloc[-1])
    print(f"Precio actual BBVA: {precio_actual:.2f}€")
    
    # RSI simplificado
    delta = precios.diff()
    ganancias = delta.where(delta > 0, 0).rolling(14).mean()
    perdidas = -delta.where(delta < 0, 0).rolling(14).mean()
    rsi = 100 - (100 / (1 + ganancias/perdidas))
    rsi_actual = float(rsi.iloc[-1])
    print(f"RSI actual BBVA: {rsi_actual:.1f}")
    
    # Interpretaciones
    if precio_actual > sma50:
        print("Tendencia: ALCISTA (precio > SMA50)")
    else:
        print("Tendencia: BAJISTA (precio < SMA50)")
    
    if rsi_actual > 70:
        print("RSI: SOBRECOMPRADO")
    elif rsi_actual < 30:
        print("RSI: SOBREVENDIDO")
    else:
        print("RSI: NEUTRAL")

In [51]:
# O SOLO EL EJEMPLO RÁPIDO:
ejemplo_rapido_bbva()

=== EJEMPLO RÁPIDO CON BBVA ===
Rentabilidad diaria promedio BBVA: 0.050%
Volatilidad anualizada BBVA: 34.48%
SMA50 actual BBVA: 14.22€
Precio actual BBVA: 15.49€
RSI actual BBVA: 38.8
Tendencia: ALCISTA (precio > SMA50)
RSI: NEUTRAL


# Análisis BBVA
* Rentabilidad diaria promedio: 0.050% (positiva, buena señal)
* Volatilidad anualizada: 34.48% (alta, típica del sector bancario)
* Precio actual: 15.49€ vs SMA50: 14.22€
* RSI: 38.8 (neutral, entre 30-70)
* Tendencencia: ALCISTA (precio está por encima de SMA50)

# Interpretación financiera: 
* BBVA está en tendencia alcista - el precio actual (15.49€) está un 8.9% por encima de la media de 50 dias (14.22€)
* RSI neutral (38.8) - no está ni sobrecomprado ni sobrevendido, hay espacio para movimiento en ambas direcciones
* Volatilidad del 34.48% - es alta pero normal para el sector bancario español
* Rentabilidad positiva - gana en promedio 0.050%

In [52]:
# EJECUTAR ANÁLISIS COMPLETO:
analisis_completo_indicadores()


ANÁLISIS TÉCNICO COMPLETO: BBVA

=== RENTABILIDADES DE BBVA ===
Rentabilidad diaria promedio: 0.050%
Mejor día: 22.03%
Peor día: -16.18%
Rentabilidad mensual promedio: 1.11%
Rentabilidad anual promedio: 13.71%
Rentabilidad total del período: 186.27%
Rentabilidad anualizada: 6.84%

=== VOLATILIDAD DE BBVA ===
Volatilidad diaria: 2.172%
Volatilidad anualizada: 34.48%
Volatilidad rolling 30 días (última): 31.94%
Volatilidad High-Low promedio: 2.73%

=== MEDIAS MÓVILES DE BBVA ===
Precio actual: 15.49€
SMA 20: 15.88€
SMA 50: 14.22€
SMA 200: 12.05€
Señal: LATERAL (señales mixtas)

=== RSI DE BBVA ===
RSI actual: 38.8
Interpretación: NEUTRAL (rango normal)
RSI promedio: 52.1
Veces sobrecomprado (>70): 728 días
Veces sobrevendido (<30): 378 días

ANÁLISIS TÉCNICO COMPLETO: Santander

=== RENTABILIDADES DE Santander ===
Rentabilidad diaria promedio: 0.038%
Mejor día: 23.22%
Peor día: -19.89%
Rentabilidad mensual promedio: 0.79%
Rentabilidad anual promedio: 9.83%
Rentabilidad total del período

## RANKING BANCARIO
# 1. Bankinter
* Rentabilidad: anualizada: 18.01% (la más alta entre todas las entidades estudiadas)
* Rentabilidad total: 1.292% en 15 años
* Mejor banco para inversión a largo plazo

# 2. Unicaja
* Rentabilidad anualizada: 12.55%
* Menor correlación con otros bancos (0.596-0.642)
* Mejor opción de diversificación

# 3. CaixaBank
* Rentabilidad anualizada: 11.85%
* Volatilidad más baja (32.99%)
* Balance rentabilidad-riesgo sólido

# 4. BBVA y Santander
* Rentabilidad BBVA: 6.84% anualizada
* Rentabilidad Santander: 3.65% anualizada
* Sorprendentemente bajo para bancos tan grandes

# 5. Sabadell
* Rentabilidad anualizada: 5.78%
* Volatilidad más alta (40.21%)
* Peor relación riesgo-rentabilidad


## ANÁLISIS DE RIESGO
# Riesgo sistémico alto (0.705 correlación promedio):
* Los bancos españoles se mueven muy similares
* Diversificar solo entre bancos españoles no reduce riesgo
* En crisis, todos caerán juntos

# Volatilidades similares (~35%):
* Sector homogéneo en términos de riesgo
* CaixaBank ligeramente menos volátil
* Sabadell el más arriesgado

## SITUACIÓN TÉCNICA ACTUAL
# Solo Santander en tendencia alcista clara:
* Precio > SMA20 > SMA50
* RSI neutral (57.8) con espacio para subir

# Resto en situación lateral:
* Señales mixtas
* Precios cerca de medias móviles
* RSI neutral en todos

## IMPLICACIONES PARA INVERSIÓN
# Si tuviera que elegir uno:
* Bankinter para máxima rentabilidad histórica
* CaixaBank para balance riesgo-rentabilidad
* Unicaja para diversificación

# Lo que NO funciona:
* Diversificar solo entre bancos españoles (correlación alta)
* Confiar en que BBVA/Santander rindan más por ser grandes
* Ignorar que Sabadell ha sido consistentemente peor

# Para trabajar en banca:
* Bankinter demuestra mejor gestión histórica
* CaixaBank muestra estabilidad
* BBVA/Santander necesitan explicar su bajo rendimiento relativo

Estos datos me sugieren que el tamaño de la entidad no garantiza mejor performance en el sector bancario español


💡 QUÉ PUEDES EXTRAER DE ESTA INFORMACIÓN
1. Análisis de Riesgo (Hecho):

Volatilidad histórica → Qué banco es más/menos arriesgado
VaR (Value at Risk) → Pérdida máxima esperada
Drawdowns → Caídas máximas desde picos

2. Análisis de Rentabilidad:

Performance comparativo → Qué banco rinde mejor
Sharpe Ratio → Mejor relación rentabilidad-riesgo
Períodos de sobre/bajo-performance

3. Análisis de Correlaciones:

Diversificación → Si se mueven igual (alta correlación = mayor riesgo sistémico)
Liderazgo sectorial → Qué banco marca tendencias
Impacto de crisis → Cómo reaccionan en conjunto

4. Análisis Temporal:

Estacionalidad → Patrones por meses/trimestres
Ciclos económicos → Performance en recesiones/expansiones
Eventos específicos → Impacto de crisis financieras (2008, COVID, etc.)

5. Señales de Trading:

Tendencias → Alcistas/bajistas
Soportes y resistencias → Niveles clave de precio
Momentum → Aceleración/desaceleración de movimientos

🎯 APLICACIONES PRÁCTICAS PARA FINANZAS
Gestión de Riesgo:

Construcción de portfolios diversificados
Límites de exposición por entidad
Modelos de estrés testing

Análisis de Inversión:

Selección de activos
Timing de entrada/salida
Benchmarking de performance

Análisis Sectorial:

Salud del sector bancario español
Comparación con benchmarks internacionales
Identificación de líderes/rezagados

## ANÁLISIS DE RIESGO
1. Valor en riesgo (VaR - Value at Risk): estima la pérdida máxima que se espera en un periodo de tiempo con un cierto nivel de confianza (en este caso aplicamos 95%).
2. Déficit Esperado (CVaR - Conditional Value at Risk): es la pérdida promedio esperada, dado que se utiliza el nivel del VaR como referencia a superar.
3. Maximum Drawdown (MDD): es la mayor caída que hasufrido una inversión, un fondo o una cartear de inversión desde un valor máximo hasta su valor mínimo, antes de que la inversión sea recuperada con un máximo nuevo.
        * Mide el riesgo a la baja y la resiliencia de una inversión
        * Es una medida de la peor serie de pérdidas consecutivas


